tests/virt_autotest/virsh_virtual_network.pm

```diff
@@ -0,0 +1,126 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify                                  
+# it under the terms of the GNU General Public License as published by                                  
+# the Free Software Foundation; either version 2 of the License, or                                     
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,                                       
+# but WITHOUT ANY WARRANTY; without even the implied warranty of                                        
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the                                         
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along                               
+# with this program; if not, see <http://www.gnu.org/licenses/>.                                        
+
+# Summary: Managing Virtual Networks with virsh
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+
+#NAT BASED NETWORK
+my $vnet_nated = <<'EOF';
```

Comment:
You declare "my" variable outside subroutines. And you only have one subroutine at the current stage.

Reply:
Cool, thanks for your comments.